src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -137,6 +138,25 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
     api_instance = kube_client.CoreV1Api()
     node_api_response = utils.validate_node_api_response(api_instance, None)
 
+    # Pre onboarding checks
+    try:
+        kubectl_client_location = install_kubectl_client()
+        helm_client_location = install_helm_client()
+        diagnostic_checks = "Failed"
+        batchv1_api_instance = kube_client.BatchV1Api()
+        # Performing cluster-diagnostic-checks
+        diagnostic_checks = precheckutils.fetch_diagnostic_checks_results(api_instance, batchv1_api_instance, helm_client_location, kubectl_client_location, kube_config, kube_context, location, http_proxy, https_proxy, no_proxy, proxy_cert)
+
+    except Exception as e:
+        logger.warning("An exception has occured while trying to execute pre-onboarding diagnostic checks : {}".format(str(e)))
```

Comment:
Add telemetry.set_exceptioon here . Also, we need to break and return here, please add that